Fix for PR #4314

boltex opened the pull request:
This PR aims at replacing completely @edreamleo 's PR #4319 (which reverted my improper PR #4314 ) By removing the problematic code, and its additional user setting in leoSettings completely. 

As @edreamleo justly noted, that problematic PR contained an error-prone 'hack' which tried to fake the javascript behavior of allowing the 'return' keyword at the top level of an evaluated piece of code (with the 'exec' python function). 

Lessons learned: I should never do non-pythonic stuff like that in Leo codebase... (I was just too excited of figuring out a hack to return values with the 'return' keyword at the top level _and I removed the 'draft' flag of this PR way too early,_ having it merged into devel before I could realize my mistakes!) This was beyond silly and I hope to clean this all up myself with this PR.

This PR also instead adresses properly my original intent of simply fixing 'doCommandByName', which did not return the value of the command called. (e.g. some commands like 'convert-blanks' and many others return something) This was done by simply adding a 'return' on the last line of _commander_command_wrapper_ in the CommanderCommand class.

A single thing remain from my original idea: The (safe and non-hacky) pythonic way to grab something programmatically from code that ran with the 'exec' function which is by capturing a 'result' named variable on the global scope after 'exec' call. 

So if this PR is accepted, the problematic code and its associated setting will be removed completely, the only two things that will be in devel would be:

1. The fix to doCommandByName to capture the returned value of existing **official commands** from Leo's codebase.
2. Having **user commands** (script in @command nodes, @button nodes, ect..) to be able to return something by setting a global named 'result'.

I've also added unit tests for those two concepts.

I will soon add a paragraph about this in the documentation, and will consult the google-group forum with a demo about those two concepts before moving on...

...and will keep this PR as draft until it's ready ! 😅

edreamleo commented:
@boltex I too am responsible for the defects of the previous PR. Not to worry, it looks like all will soon be well.

boltex commented:
@edreamleo I've cleaned up a bit and I've also added to the documentation and removed obsolete stuff related to the return values of commands ran 'by name'.  

(I've only modified the Leo files, not generated the website files which can be done later in the devel branch before releasing the next version.)
